### PR TITLE
Fix EZP-25817: Invalid subtree path when assigning role with subtree limitation (master)

### PR DESCRIPTION
**Pull request by glye**
> Fix https://jira.ez.no/browse/EZP-25817
> Status: Ready for review

Throws an exception when the subtree path does not have the required trailing slash. This breaks PlatformUI, where the real fix should be made. This should be merged only with master and only when PlatformUI is fixed.

The 6.3 fix will simply append the slash, to avoid breaking things: https://github.com/ezsystems/ezpublish-kernel/pull/1670

- [x] Fix breaking tests
- [x] Mysql update
- [x] Postgres update

### CODE REVIEW

**Comment by andrerom** [comment]
+1 when it passes, but should be merged after the 6.3 one so conflicts are handled in favour of this one.

**Comment by glye** [comment]
Rebased after 6.3 fix merged.

**Comment by andrerom** [comment]
Side: This one can't go in before UI fix is in place as well.

**Comment by yannickroger** [comment]
+1 on the fix with @andrerom 's Note2 (update sql)

I am currently working on the PlatformUI fix, I will add a comment here with the link when it is ready.

**Comment by glye** [comment]
Added mysql update sql - please verify file name vs versions (6.4alpha1)
Will add postgres too.

**Comment by glye** [comment]
Added postgres update sql - based on docs, not yet tested.

**Comment by yannickroger** [comment]
You can now check https://github.com/ezsystems/PlatformUIBundle/pull/619

It seems that after adding the limitation the list is not refreshed, I will create a jira issue about it (after checking there isn't one already).

**Comment by glye** [comment]
@bdunogier Can you please verify the filenames (i.e. release versions) of the sql updates?

**Comment by bdunogier** [comment]
I'm not that comfortable with our database files naming, gotta admit.

I don't mind having the final version number if the database dump. Should we just rename it to the final name ?

**Comment by glye** [comment]
Ok with me. You want to take it out of the "unstable" directory too then?

**Comment by yannickroger** [comment]
PlatformUI fix merged, this one is safe to go.

**Comment by glye** [comment]
Renamed db updates to final name and moved out of "unstable" dirs.

File: data/update/mysql/dbupdate-6.3.0-to-6.4.0.sql [hunk 1]
```diff
@@ -3,7 +3,7 @@ SET default_storage_engine=InnoDB;
 UPDATE ezsite_data SET value='6.4.0' WHERE name='ezpublish-version';
 
 --
--- EZP-25880: Make ezuser.login case in-sensitive across databases, using case in-sensitive index
+-- EZP-25817: Invalid subtree path when assigning role with subtree limitation
 --
 
-ALTER TABLE ezuser DROP COLUMN login_normalized, ADD UNIQUE KEY ezuser_login (login);
```
Comment (andrerom) [inline data/update/mysql/dbupdate-6.3.0-to-6.4.0.sql]: please don't remove

**Comment by glye** [comment]
Fixed broken previous commit.

File: data/update/postgres/dbupdate-6.3.0-to-6.4.0.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+-- Set storage engine schema version number
+UPDATE ezsite_data SET value='6.4.0' WHERE name='ezpublish-version';
+
+--
+-- EZP-25817: Invalid subtree path when assigning role with subtree limitation
+--
+
+UPDATE ezuser_role SET limit_value = CONCAT(limit_value, '/') WHERE limit_identifier = 'Subtree' AND limit_value ~ '[[:digit:]]$';
```
Comment (andrerom) [inline data/update/postgres/dbupdate-6.3.0-to-6.4.0.sql]: missing changes from EZP-25880, see 5.4 => 6.4

Comment (glye) [inline data/update/postgres/dbupdate-6.3.0-to-6.4.0.sql]: Added, thanks.

**Pull request merged**